src/App.tsx

```diff
@@ -1,56 +1,79 @@
+  let timerId: number;
+
+  return (...args: string[]) => {
+    clearTimeout(timerId);
+    timerId = window.setTimeout(f, delay, ...args);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selected, setSelected] = useState<Person | null>(null);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const selectPerson = useCallback((person: Person) => {
+    setSelected(person);
+    setQuery('');
+    setAppliedQuery('');
+    setShowSuggestions(state => !state);
+  }, [setSelected, setQuery, setAppliedQuery]);
+
+  const visiblePeope = useMemo(() => {
+    return peopleFromServer.filter(
+      person => person.name
+        .toLowerCase().includes(appliedQuery.toLowerCase().trim()),
+    );
+  }, [peopleFromServer, appliedQuery]);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selected ? (
+          `${selected.name} (${selected.born} = ${selected.died})`
+        ) : (
+          'No selected person'
+        )}
       </h1>
 
-      <div className="dropdown is-active">
+      <div className={classNames(
+        'dropdown',
+        { 'is-active': !showSuggestions },
+      )}
+      >
         <div className="dropdown-trigger">
           <input
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={event => {
+              setQuery(event.target.value);
+              applyQuery(event.target.value);
+              setShowSuggestions(false);
+            }}
           />
         </div>
 
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
-          </div>
-        </div>
+        {appliedQuery && (
```

Comment:
additional state showSuggestions is redundant.
enought was to add this condition